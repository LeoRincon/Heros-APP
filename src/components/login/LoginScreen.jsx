import React, { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { types } from '../../types/index';
import { AuthContext } from '../../auth/AuthContext';

export const LoginScreen = () => {
  let navigate = useNavigate();
  const { dispatch } = useContext(AuthContext);

  const handleLogin = () => {
    const action = {
      type: types.login,
      payload: { name: 'Juan' },
    };

    dispatch(action);

    const lastPath = localStorage.getItem('lastPath') || '/';
    navigate(lastPath, { replace: true });
  };

  return (
    <div className='container mt-5'>
      <h2>Login</h2>
      <hr />
      <button className='btn btn-primary' onClick={handleLogin}>
        login
      </button>
    </div>
  );
};
