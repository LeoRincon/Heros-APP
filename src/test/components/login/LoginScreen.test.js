import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/index';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('test in <LoginScreen /> when is called', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={['/login']}>
        <Routes>
          <Route path='/login' element={<LoginScreen />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  test('should be displayed correctly LoginScreen', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call the login, call the navigate and dispatch with the arguments', () => {
    const handleClick = wrapper.find('button').prop('onClick');
    handleClick();

    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: { name: 'Juan' },
    });

    expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true });

    localStorage.setItem('lastPath', '/dc');
    handleClick();
    expect(mockNavigate).toHaveBeenCalledWith('/dc', { replace: true });
  });
});
