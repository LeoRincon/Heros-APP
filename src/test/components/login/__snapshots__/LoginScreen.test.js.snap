// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <LoginScreen /> when is called should be displayed correctly LoginScreen 1`] = `
<div
  className="container mt-5"
>
  <h2>
    Login
  </h2>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
